### PR TITLE
Fix duplicate rows and added scope for component name

### PR DESCRIPTION
**Pull request by asafambar**
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Comment by asafambar** [comment]
Curation Xray reports show for each direct dependencies all its blocked packages under curation. Under a direct dependency can be many packages which are similar under different paths, we want to present in the report only one occurrence for each package.
Also added scope for packge name, scope is also part of the component name uniqueness,
according to NPM official desperation of scope:
_"All npm packages have a name. Some package names also have a scope. A scope follows the usual rules for package names (URL-safe characters, no leading dots or underscores). When used in package names, scopes are preceded by an @ symbol and followed by a slash, e.g."_

File: xray/commands/curation/audit.go [hunk 1]
```diff
@@ -495,7 +509,7 @@ func getUrlNameAndVersionByTech(tech coreutils.Technology, nodeId, artiUrl, repo
 
 // The graph holds, for each node, the component ID (xray representation)
 // from which we extract the package name, version, and construct the Artifactory download URL.
-func getNameScopeAndVersion(id, artiUrl, repo, tech string) (downloadUrl, name, version string) {
+func getNameScopeAndVersion(id, artiUrl, repo, tech string) (downloadUrl, name, scope, version string) {
```
Comment (yahavi) [inline xray/commands/curation/audit.go:512]:
```suggestion
func getNpmNameScopeAndVersion(id, artiUrl, repo, tech string) (downloadUrl, name, scope, version string) {
```

File: xray/commands/curation/audit_test.go [hunk 2]
```diff
@@ -148,17 +149,21 @@ func TestGetNameScopeAndVersion(t *testing.T) {
...
 			}
 			if gotName != tt.wantName {
 				t.Errorf("getNameScopeAndVersion() gotName = %v, want %v", gotName, tt.wantName)
 			}
+			if gotScope != tt.wantScope {
+				t.Errorf("getNameScopeAndVersion() gotName = %v, want %v", gotName, tt.wantName)
+			}
 			if gotVersion != tt.wantVersion {
 				t.Errorf("getNameScopeAndVersion() gotVersion = %v, want %v", gotVersion, tt.wantVersion)
 			}
```
Comment (yahavi) [inline xray/commands/curation/audit_test.go:169]: Let's use assert.Equal to make the code easier to read. For example:
```go
assert.Equal(t, tt.wantDownloadUrl, gotDownloadUrl, "getNameScopeAndVersion() gotDownloadUrl = %v, want %v", gotDownloadUrl, tt.wantDownloadUrl)

```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: xray/commands/curation/audit.go [hunk 3]
```diff
@@ -336,13 +339,21 @@ func (ca *CurationAuditCommand) SetRepo(tech coreutils.Technology) error {
 }
 
 func (nc *treeAnalyzer) fillGraphRelations(node *xrayUtils.GraphNode, preProcessMap *sync.Map,
-	packagesStatus *[]*PackageStatus, parent, parentVersion string, isRoot bool) {
+	packagesStatus *[]*PackageStatus, parent, parentVersion string, visited map[string]struct{}, isRoot bool) {
```
Comment (yahavi) [inline xray/commands/curation/audit.go:342]: Let's use set here
```suggestion
	packagesStatus *[]*PackageStatus, parent, parentVersion string, visited datastructures.Set[string], isRoot bool) {
	for _, child := range node.Nodes {, isRoot bool) {
```

File: xray/commands/curation/audit_test.go [hunk 4]
```diff
@@ -148,19 +151,23 @@ func TestGetNameScopeAndVersion(t *testing.T) {
...
 			if gotName != tt.wantName {
-				t.Errorf("getNameScopeAndVersion() gotName = %v, want %v", gotName, tt.wantName)
+				assert.Equal(t, tt.wantName, gotName, "getNpmNameScopeAndVersion() gotName = %v, want %v", gotName, tt.wantName)
+			}
+			if gotScope != tt.wantScope {
+				assert.Equal(t, tt.wantScope, gotScope, "getNpmNameScopeAndVersion() gotScope = %v, want %v", gotScope, tt.wantScope)
 			}
 			if gotVersion != tt.wantVersion {
-				t.Errorf("getNameScopeAndVersion() gotVersion = %v, want %v", gotVersion, tt.wantVersion)
+				assert.Equal(t, tt.wantVersion, gotVersion, "getNpmNameScopeAndVersion() gotVersion = %v, want %v", gotVersion, tt.wantVersion)
```
Comment (yahavi) [inline xray/commands/curation/audit_test.go:170]: Please remove the "if"s @asafambar

**Review by yahavi — APPROVED** [review]
LGTM

**Pull request merged**